extensions/src/main/java/org/pentaho/platform/plugin/action/ActionRunner.java

```diff
@@ -68,14 +70,17 @@ public ActionRunner( final IAction actionBean, final String actionUser, final Ma
   }
 
   public Boolean call() throws ActionInvocationException {
+    final String workItemUid = ActionUtil.extractUid( params );
     try {
-      return callImpl();
+      final boolean result = callImpl();
+      WorkItemLifecyclePublisher.publish( workItemUid, params, WorkItemLifecyclePhase.SUCCEEDED );
```

Comment:
@pedrofvteixeira I agree that this work is better for consistency/clarity sake, but I think it's not within the scope of this PR? Maybe we can create a jira for it?